File: Data Structures & Algorithms/two-integer-sum/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        HashSet<int> hashSet = new HashSet<int>();
        int[] ans = new int[2];

        foreach (int num in nums)
        {
            hashSet.Add(num);
        }

        for(int i = 0; i<nums.Length; i++){
            if(hashSet.Contains(target - nums[i])){
                if(Array.IndexOf(nums, target - nums[i]) > i){
                    ans[0] = i;
                    ans[1] = Array.IndexOf(nums, target - nums[i]);
                } else {
                    ans[1] = i;
                    ans[0] = Array.IndexOf(nums, target - nums[i]);
                }
            }
        }

        return ans;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int s = 0, e = numbers.length - 1;

        while(s<e){
            int sum = numbers[s] + numbers[e];
            
            if(sum == target){
                return new int[]{s + 1, e + 1}; // 1-based indexing
            }
            
            if(sum > target) {
                e--;
            } else {
                s++;
            }
        }

       return new int[]{-1, -1};
    }
}
